astropy/utils/data.py

```diff
@@ -983,24 +983,42 @@ def connect_ftp(self, user, passwd, host, port, dirs, timeout):
 
 def _download_file_from_source(source_url, show_progress=True, timeout=None,
                                remote_url=None, cache=False, pkgname='astropy',
-                               http_headers=None, ftp_tls=False):
+                               http_headers=None, ftp_tls=None):
     from astropy.utils.console import ProgressBarOrSpinner
 
+    if timeout == 0:
+        raise urllib.error.URLError(
+            f"URL {remote_url} was supposed to be downloaded but timeout was set to 0; "
+            f"if this is unexpected check the astropy config file for the option "
+            f"remote_timeout")
+
     if remote_url is None:
         remote_url = source_url
     if http_headers is None:
         http_headers = {}
 
+    if ftp_tls is None and urllib.parse.urlparse(remote_url).scheme == "ftp":
```

Comment:
So, in this PR `ftp_tls` is removed from public API without any deprecation period. I think this is only acceptable if #9964 is not really released yet, so I'll have to defer to @eteq .

p.s. I thought I commented about this in my review earlier but I don't see it. But if you getting duplicate messages about this, I apologize.

Reply:
I checked this - ftp_tls would be added in the same release these changes go in. So this just amounts to not adding it to the public API at all..